mypy/checkexpr.py

```diff
@@ -6210,7 +6210,7 @@ def visit_type_var_expr(self, e: TypeVarExpr) -> Type:
         ):
             if not is_subtype(p_default, e.upper_bound):
                 self.chk.fail("TypeVar default must be a subtype of the bound type", e)
-            if e.values and not any(p_default == value for value in e.values):
+            if e.values and not any(is_same_type(p_default, value) for value in e.values):
                 self.chk.fail("TypeVar default must be one of the constraint types", e)
```

Comment:
It's a bit strange to have this logic duplicated! Is this because PEP 695 doesn't use `TypeVarExpr` so checkexpr will never hit this?

Reply:
Yes, I wasn't able to get rid of one of these checks, but made a note to myself to try harder later:)